### PR TITLE
Add an additional event FoundLocal.

### PR DESCRIPTION
**Pull request by rklaehn**
## Description

This event needs to be considered when computing how much data needs to be downloaded in case of resume.

Also change hashseq download to assume that at least the hashseq itself is complete. We currently do not support resume if the download of the hashseq itself was interrupted.

## Notes & open questions

1. I think this provides the required info to get good progress bars even in case of resume. But I don't know for sure until I have hacked it into sendme.
2. Sendme does 2 roundtrips and has accurate progress bars. Iroh does 1 request and has a 2 level progress bar because the total size is not known in advance. Should we change iroh to do what sendme does?

## Change checklist

- [x] Self-review.
- [x] Documentation updates if relevant.
- [ ] Tests if relevant.

### CODE REVIEW

**Comment by rklaehn** [comment]
Note for reviewers: there are not as many changes as it looks like from the PR.

Basically I changed
```rust
let finishing = if let Some(entry) = db.get(root_hash) {
```

to
```rust
    let finishing =
        if let PossiblyPartialEntry::Complete(entry) = db.get_possibly_partial(root_hash) {
```

and that changed the indentation level for all the code below.

File: iroh-bytes/src/provider.rs [hunk 1]
```diff
@@ -155,6 +155,17 @@ pub enum AddProgress {
...
 #[derive(Debug, Serialize, Deserialize)]
 pub enum DownloadProgress {
+    /// Data was found locally.
+    FoundLocal {
+        /// child offset
+        child: u64,
+        /// The hash of the entry.
+        hash: Hash,
+        /// The size of the entry in bytes.
+        size: u64,
```
Comment (dignifiedquire) [inline iroh-bytes/src/provider.rs:165]: is this the full size, or the size that resides locally?

Comment (rklaehn) [reply iroh-bytes/src/provider.rs:165]: No, it is the full size. That is one thing we know as soon as we get any data. To compute the size that is local you need to use this and valid_ranges.

Comment (dignifiedquire) [reply iroh-bytes/src/provider.rs:165]: I can see how the ranges could be useful, if you want to show a visualization which pieces of a file are already available on disk. But the more common use case seems to be just showing `x/y` has already been downloaded.

Comment (dignifiedquire) [reply iroh-bytes/src/provider.rs:165]: so maybe at least have a simple helper method on getting the local size?

**Pull request merged**